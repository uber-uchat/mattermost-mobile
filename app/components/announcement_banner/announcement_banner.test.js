// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

import AnnouncementBanner from './announcement_banner.js';

jest.useFakeTimers();

describe('AnnouncementBanner', () => {
    const baseProps = {
        actions: {
            dismissBanner: jest.fn(),
        },
        allowDismissal: true,
        bannerColor: '#ddd',
        bannerDismissed: false,
        bannerEnabled: true,
        bannerText: 'Batter Text',
        bannerTextColor: '#fff',
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <AnnouncementBanner {...baseProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should call actions.dismissBanner on handleDismiss', () => {
        const actions = {dismissBanner: jest.fn()};
        const props = {...baseProps, actions};
        const wrapper = shallow(
            <AnnouncementBanner {...props}/>
        );

        wrapper.instance().handleDismiss();
        expect(actions.dismissBanner).toHaveBeenCalledTimes(1);
        expect(actions.dismissBanner).toHaveBeenCalledWith(props.bannerText);
    });
});