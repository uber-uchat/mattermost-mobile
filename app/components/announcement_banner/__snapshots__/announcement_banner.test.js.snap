// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnouncementBanner should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnnouncementBanner
    actions={
      Object {
        "dismissBanner": [MockFunction],
      }
    }
    allowDismissal={true}
    bannerColor="#ddd"
    bannerDismissed={false}
    bannerEnabled={true}
    bannerText="Batter Text"
    bannerTextColor="#fff"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "flex": 1,
                "fontSize": 14,
                "marginRight": 5,
              },
              Object {
                "color": "#fff",
              },
            ]
          }
        >
          Batter Text
        </Text>
        <Icon
          allowFontScaling={false}
          color="#fff"
          name="info"
          size={16}
        />
      </TouchableOpacity>,
      "style": Array [
        Object {
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#ddd",
          "height": 0,
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 14,
                  "marginRight": 5,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Batter Text
          </Text>,
          <Icon
            allowFontScaling={false}
            color="#fff"
            name="info"
            size={16}
          />,
        ],
        "onPress": [Function],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Batter Text",
            "ellipsizeMode": "tail",
            "numberOfLines": 1,
            "style": Array [
              Object {
                "flex": 1,
                "fontSize": 14,
                "marginRight": 5,
              },
              Object {
                "color": "#fff",
              },
            ],
          },
          "ref": null,
          "rendered": "Batter Text",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "color": "#fff",
            "name": "info",
            "size": 16,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 14,
                  "marginRight": 5,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Batter Text
          </Text>
          <Icon
            allowFontScaling={false}
            color="#fff"
            name="info"
            size={16}
          />
        </TouchableOpacity>,
        "style": Array [
          Object {
            "overflow": "hidden",
            "paddingHorizontal": 10,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#ddd",
            "height": 0,
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 14,
                    "marginRight": 5,
                  },
                  Object {
                    "color": "#fff",
                  },
                ]
              }
            >
              Batter Text
            </Text>,
            <Icon
              allowFontScaling={false}
              color="#fff"
              name="info"
              size={16}
            />,
          ],
          "onPress": [Function],
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Batter Text",
              "ellipsizeMode": "tail",
              "numberOfLines": 1,
              "style": Array [
                Object {
                  "flex": 1,
                  "fontSize": 14,
                  "marginRight": 5,
                },
                Object {
                  "color": "#fff",
                },
              ],
            },
            "ref": null,
            "rendered": "Batter Text",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": false,
              "color": "#fff",
              "name": "info",
              "size": 16,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
